cloudamqp/resource_cloudamqp_security_firewall.go

```diff
@@ -37,15 +42,23 @@ func resourceSecurityFirewall() *schema.Resource {
 								Type:         schema.TypeString,
 								ValidateFunc: validateServices(),
 							},
-							Description: "Pre-defined services 'AMQP', 'AMQPS', 'HTTPS', 'MQTT', 'MQTTS', 'STOMP', 'STOMPS', "+
-																								 "'STREAM', 'STREAM_SSL'",
+							Description: "Pre-defined services 'AMQP', 'AMQPS', 'HTTPS', 'MQTT', 'MQTTS', 'STOMP', 'STOMPS', " +
+								"'STREAM', 'STREAM_SSL'",
 						},
 						"ports": {
 							Type:     schema.TypeList,
 							Optional: true,
 							Elem: &schema.Schema{
-								Type:         schema.TypeInt,
-								ValidateFunc: validation.IntBetween(0, 65554),
+								Type: schema.TypeInt,
+								ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+									v := val.(int)
+									if v < 0 || v > 65554 {
+										errs = append(errs, fmt.Errorf("%q must be between 0 and 65554, got: %d", key, v))
+									} else if validateServicePort(v) {
+										errs = append(errs, fmt.Errorf("port %d, pre-defined as service %q", v, portToService(v)))
```

Comment:
```suggestion
										errs = append(errs, fmt.Errorf("Port %d found in \"ports\", needs to be added as %q in \"services\" instead", v, portToService(v)))
```

More actionable message:

````
$ terraform plan -out=plan
╷
│ Error: Port 5671 found in "ports", needs to be added as "AMQPS" in "services" instead
│
│   with cloudamqp_security_firewall.firewall_settings,
│   on main.tf line 20, in resource "cloudamqp_security_firewall" "firewall_settings":
│   20: resource "cloudamqp_security_firewall" "firewall_settings" {
│
╵
╷
│ Error: Port 443 found in "ports", needs to be added as "HTTPS" in "services" instead
│
│   with cloudamqp_security_firewall.firewall_settings,
│   on main.tf line 20, in resource "cloudamqp_security_firewall" "firewall_settings":
│   20: resource "cloudamqp_security_firewall" "firewall_settings" {
│
╵
````

Reply:
Great, way better.